File: movie-infrastructure-module/src/main/java/com/app/movie/repository/SeatJpaRepository.java
package com.app.movie.repository;

import com.app.movie.entity.SeatEntity;
import jakarta.persistence.LockModeType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Lock;

import java.util.Optional;

public interface SeatJpaRepository extends JpaRepository<SeatEntity, Long> {

//    @Lock(LockModeType.PESSIMISTIC_WRITE)
    Optional<SeatEntity> findById(Long id);

}

File: infrastructure/mysql/src/main/kotlin/com/hanghe/redis/seat/SeatJpaRepository.kt
package com.hanghe.redis.seat

import com.hanghe.redis.movie.seat.SeatEntity
import org.springframework.data.jpa.repository.JpaRepository

interface SeatJpaRepository : JpaRepository<SeatEntity, Long> {

    fun findByTheaterIdAndIdIn(theaterId: Long, seatIds: List<Long>): List<SeatEntity>
}
